elastic/kibana
# [Console Monaco] Add theme for yaml and text mode in output

Addresses https://github.com/elastic/kibana/issues/179905

## Summary

This PR adds themes for the Console output panel when it is in YAML or Text mode. The main theme features that are specified for these two main modes are colors (e.g. background color).

**How to test:**

1. Create a `config/kibana.dev.yml` file (if one doesn't exist already) and add the line: `console.dev.enableMonaco: true`
2. Change the condition in https://github.com/elastic/kibana/blob/b8c13babf1c24a2a50668bc5ed647ce87deec5f0/src/plugins/console/public/application/containers/editor/editor.tsx#L76 to `!isMonacoEnabled` to display the old Ace editor in the request panel for now so that we can test sending a request.
3. Send a request and verify that the theme for json mode hasn't changed.
4. Send a request with `?format=yaml` parameter (e.g. `GET _all?format=yaml`) and verify that the theme (apart from the text colors) is the same as in json.

<img width="705" alt="Screenshot 2024-04-05 at 10 24 47" src="https://github.com/elastic/kibana/assets/59341489/8de43eb2-068b-4f5d-84d5-2a54ebd4c47f">


5. The output also supports text output data. To test this format, comment out line 30 and add the following code in line 34 in `src/plugins/console/public/application/containers/editor/monaco/monaco_editor_output.tsx`:

```
const data = [
      {
        response: { value: 'Hello World!' },
      },
    ];
```

Then reload the Console page and verify that the theme (apart from the text colors) is the same as in json:

<img width="705" alt="Screenshot 2024-04-05 at 10 26 11" src="https://github.com/elastic/kibana/assets/59341489/96fe8bf5-2264-4860-8b29-efa5ad33a015">
